pyspecdata/core.py

```diff
@@ -1333,14 +1334,27 @@ def show_array(x, indent=""):
             else:
                 return x
 
+        if self.data.size < 2:
+            val = self.data.item()
+            err = self.get_error()
+            if err is not None:
+                err = err.item()
+                oom_err = int(np.floor(np.log10(err)))  # int takes floor
+                oom_val = int(np.floor(np.log10(val)))  # int takes floor
+                retval = (
+                    "%#0." + str(oom_val - oom_err + 1) + "g ± %#0.2g"
+                ) % (val, err)
+            else:
+                retval = "%#0.5g" % val
+            myunits = self.get_units()
+            if myunits is not None:
+                retval += f" {Q_(myunits).units:~P}"
+            return retval
```

Comment:
ooo fancy I like this so now when we get the returned value with error it has units in the str